File: frontend/src/types/match.ts
export type MatchStatus = "pending" | "running" | "completed" | "failed";
export type MarketType = "random" | "trending" | "ranging";
export type MarketSource =
  | "simulated"
  | "coingecko_historical"
  | "coingecko_realtime";

export interface CoinInfo {
  id: string;
  symbol: string;
  name: string;
}

export interface MatchParticipant {
  strategy_id: string;
  strategy_name?: string;
  final_value?: number;
  return_pct?: number;
  total_trades: number;
  win_trades: number;
  rank?: number;
  max_drawdown?: number;
  sharpe_ratio?: number;
  value_history?: number[];
}

export interface MatchConfig {
  initial_capital: number;
  trading_pair: string;
  timeframe: string;
  duration_steps: number;
  market_type?: MarketType;
  market_source?: MarketSource;
  coin_id?: string;
}

export interface ExecutionEntry {
  step: number;
  strategy_id: string;
  action: { type: string; asset?: string; amount?: number };
  portfolio: {
    cash: number;
    positions: Record<string, number>;
    total_value: number;
  };
  price: number;
}

export interface MatchLogEntry {
  step: number;
  data: { logs: ExecutionEntry[] };
}

export interface Match {
  id: string;
  status: MatchStatus;
  error_message?: string;
  config: MatchConfig;
  created_at: string;
  start_time?: string;
  end_time?: string;
  participants: MatchParticipant[];
  participants_count?: number;
  logs?: MatchLogEntry[];
}

export interface RunMatchRequest {
  strategy_ids: string[];
  market_type: MarketType;
  market_source: MarketSource;
  coin_id: string;
  duration_steps: number;
  initial_capital: number;
}

File: backend/app/models/match.py
from pydantic import BaseModel, Field
from typing import List, Dict, Any, Optional
from datetime import datetime
from enum import Enum

class MatchStatus(str, Enum):
    """比赛状态"""
    PENDING = "pending"
    BETTING = "betting"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"

class Action(BaseModel):
    """交易指令"""
    type: str = Field(..., pattern="^(buy|sell|hold)$")
    asset: str = "ETH"
    amount: float = Field(default=0.0, ge=0)

    class Config:
        json_schema_extra = {
            "example": {
                "type": "buy",
                "asset": "ETH",
                "amount": 1000.0
            }
        }

class MatchConfig(BaseModel):
    """比赛配置"""
    initial_capital: float = 10000.0
    trading_pair: str = "ETH/USDC"
    timeframe: str = "5m"
    duration_steps: int = 100

class MatchResult(BaseModel):
    """比赛结果"""
    strategy_id: str
    final_value: float
    return_pct: float
    total_trades: int
    win_trades: int
    rank: int
    max_drawdown: float = 0.0   # 最大回撤 (%)
    sharpe_ratio: float = 0.0   # 夏普率

class Match(BaseModel):
    """比赛模型"""
    id: str
    status: MatchStatus = MatchStatus.PENDING
    config: MatchConfig
    strategy_ids: List[str]

    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None

    results: List[MatchResult] = Field(default_factory=list)
    execution_log: List[Dict[str, Any]] = Field(default_factory=list)

    created_at: datetime = Field(default_factory=datetime.now)
